python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,250 @@
+                self.tarts.alignnet_model.alignnet.device_val = torch.device("cpu")
+            if hasattr(self.tarts.wavenet_model.wavenet, 'device_val'):
+                self.tarts.wavenet_model.wavenet.device_val = torch.device("cpu")
+
+            # Ensure all CNN models within the sub-models are also on CPU
+            if hasattr(self.tarts.alignnet_model.alignnet, 'cnn'):
+                self.tarts.alignnet_model.alignnet.cnn = self.tarts.alignnet_model.alignnet.cnn.cpu()
+            if hasattr(self.tarts.wavenet_model.wavenet, 'cnn'):
+                self.tarts.wavenet_model.wavenet.cnn = self.tarts.wavenet_model.wavenet.cnn.cpu()
+        else:
+            self.tarts.to("cuda")
+
+    def calc_exposure(self, exposure: Any) -> np.ndarray:
+        """
+        Calculate the Zernike coefficients for an exposure.
+        """
+        with torch.no_grad():
+            pred = self.tarts.deploy_run(exposure)
+        return pred
+
+    def empty(self, qualityTable: Optional[QTable] = None) -> pipeBase.Struct:
+        """Return empty results if no donuts are available. If
+        it is a result of no quality donuts we still include the
+        quality table results instead of an empty quality table.
+
+        Parameters
+        ----------
+        qualityTable : astropy.table.QTable
+            Quality table created with donut stamp input.
+
+        Returns
+        -------
+        lsst.pipe.base.Struct
+            Empty output tables for zernikes. Empty quality table
+            if no donuts. Otherwise contains quality table
+            with donuts that all failed to pass quality check.
+        """
+        qualityTableCols = [
+            "SN",
+            "ENTROPY",
+            "ENTROPY_SELECT",
+            "SN_SELECT",
+            "FINAL_SELECT",
+            "DEFOCAL_TYPE",
+        ]
+        if qualityTable is None:
+            donutQualityTable = QTable({name: [] for name in qualityTableCols})
+        else:
+            donutQualityTable = qualityTable
+        return pipeBase.Struct(
+            outputZernikesRaw=np.atleast_2d(np.full(len(self.nollIndices), np.nan)),
+            outputZernikesAvg=np.atleast_2d(np.full(len(self.nollIndices), np.nan)),
+            zernikes=self.initZkTable(),
+            donutQualityTable=donutQualityTable,
+        )
+
+    def initZkTable(self) -> QTable:
+        """Initialize an empty Zernike coefficient table.
+
+        Returns
+        -------
+        astropy.table.QTable
+            Empty table with columns for Zernike coefficients.
+        """
+        # Create columns for each Noll index
+        columns: dict[str, list] = {}
+        for noll_idx in self.nollIndices:
+            columns[f"Z{noll_idx}"] = []
+
+        return QTable(columns)
+
+    @timeMethod
+    def run(
+        self,
+        ExtraExposure: Any,
+        IntraExposure: Any,
+    ) -> pipeBase.Struct:
+
+        pred_intra = self.calc_exposure(IntraExposure) #gives microns
```

Comment:
I think this needs to be modified. You are assuming you are passing IntraExposure and ExtraExposure, but most of the times we run in CWFS mode (that is, we have one exposure, and we are using that exposure intra and extra-focal corners to compute the zernikes, how is this covered in your code?